chain/chunks/src/lib.rs

```diff
@@ -123,6 +128,105 @@ impl RequestPool {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Seal {
+    part_ords: HashSet<u64>,
+    chunk_producer: AccountId,
+    sent: DateTime<Utc>,
+}
+
+impl Seal {
+    fn process(&mut self, chunk_entry: &EncodedChunksCacheEntry) -> bool {
+        let mut part_ords_to_remove = HashSet::new();
+        let mut res = true;
+        for part_ord in self.part_ords.iter() {
```

Comment:
Would `self.part_ords.retain` work here?

Reply:
Yes. How do you suggest to use it?

Reply:
Updated to:
```
        self.part_ords.retain(|part_ord| {
            if !chunk_entry.parts.contains_key(&part_ord) {
                res = false;
                true
            } else {
                false
            }
        });
```